grpc-codegen/src/main/scala/monix/grpc/codegen/GrpcCodeGenerator.scala

```diff
@@ -4,11 +4,26 @@ import com.google.protobuf.ExtensionRegistry
 import com.google.protobuf.compiler.PluginProtos.CodeGeneratorResponse
 import protocbridge.Artifact
 import protocgen.{CodeGenApp, CodeGenRequest, CodeGenResponse}
-import scalapb.compiler.{DescriptorImplicits, FunctionalPrinter, ProtobufGenerator}
+import scalapb.compiler.{DescriptorImplicits, FunctionalPrinter, GeneratorParams, ProtobufGenerator}
 import scalapb.options.Scalapb
 import monix.grpc.codegen.build.BuildInfo
 
-case class CodeGenParams(serviceSuffix: String = "Api")
+case class CodeGenParams(serviceSuffix: String = "Api", disableMetaData: Boolean = false)
+
+object CodeGenParams {
+  def fromString(params: String): (CodeGenParams, String) =
+    params
+      .split(',')
+      .map(_.trim)
+      .filter(_.nonEmpty)
+      .foldLeft[(CodeGenParams, String)]((CodeGenParams(), "")) {
+        case ((params, unknownParameters), str) =>
+          str match {
+            case "disable_metadata" => (params.copy(disableMetaData = true), str)
+            case other => (params, unknownParameters + "," + other)
```

Comment:
Oke so here is the trick if any of our parameters ends up in the `ProtobufGenerator.parseParameters` it would fail so i collect our parameters out of the parameters and then pass the rest of the parameters to ProtobufGenerator. Then both of these parameters extractors are happy.